src/repo/repository.rs

```diff
@@ -182,15 +182,24 @@ impl Repository {
             return Err(Error::RepositoryNotFound);
         }
 
-        let lock_file = fs::File::create(data_dir.join("mutex"))?;
-        if let Err(e) = lock_file.try_lock_shared() {
-            if e.raw_os_error() == fs2::lock_contended_error().raw_os_error() {
-                warn!("Blocking until the repository mutex is unlocked...");
-                lock_file.try_lock_shared()?;
-            } else {
-                return Err(e.into());
+        let readonly = fs::metadata(&data_dir)?.permissions().readonly();
```

Comment:
Rather than checking permissions, I'm thinking it would be better to check the error code for file creation. This way you avoid needing an additional check and avoid the possibility that the state changes between the two (admittedly an unlikely scenario here). Also, I'm not sure if `permissions().readonly()` would detect if it were a readonly mount for example.